README.md

```diff
@@ -246,14 +274,50 @@ mgit exec 'echo `pwd`'
 # /home/mgit/packages/organization/repository-2
 ```
 
-### save-hashes
+### commit (alias: `ci`)
+
+For every repository that contains changes which can be committed, makes a commit with these files. 
+You need to specify the message for the commit.
+
+Example:
+
+```bash
+mgit commit -- --message 'Introduce PULL_REQUEST_TEMPLATE.md.'
+
+# Executes `git commit --message 'Introduce PULL_REQUEST_TEMPLATE.md.'` command on each repository.
+# Commit will be made in repositories that "git status" returns a list if changed files (these files must be tracked by Git).
+```
+
+### merge
+
+Requires a second argument which is a branch name that will be merged to current one. You can also specify the message
+which will be added to the default git-merge message.
+
+Repositories which do not have specified branch will be ignored.
+
+Example:
+
+```bash
+# Assumptions: we are on "master" branch and "develop" branch exists.
+mgit merge develop -- --message 'These changes are required for the future release.'
+
+# Branch `develop` will be merged into `master`.
+```
+
+### save
 
 Saves hashes of packages in `mgit.json`. It allows to easily fix project to a specific state.
 
 Example:
 
 ```bash
-mgit save-hashes
+mgit save
+```
+
+If you would like to save name of branches instead of current commit, you can use an option `--branch`:
+
+```bash
+mgit save -- --branch
```

Comment:
`--branch` is the param of `mgit save`, not of `git` itself, so while this makes sense

```
mgit commit -- -m "Foo"
```

because `-m "Foo"` goes to `git`, `--branch` is a param of mgit's logic itself, so that'd be reasonable (and easier):

```
mgit save --branch
```

Reply:
I don't agree. `--branch` or `--hash` (which is the default) say how `save` command should behave (what should be collected - branch names or current commit hashes). It does not touch mgit. 

I agree that `mgit save --branch` is easier and we could consider removing the `--` separator. Then we could pass all the options directly. I'm pretty sure that we will forget which options touch mgit and which are specified only for commands. But in this case we will lost aliases.

Reply:
> It does not touch mgit.

Do you use git like this?

```
git commit -- --message
```

No, you don't. Despite pretty much the same look to the end user (`mgit whatever` looks like `git whatever`) those tools work differently now.

And once again – as a user of this tool I should not need to care about how it's implemented internally. If the implementation changes (we remove the concept of tasks), will we change the params list?

Reply:
Git works on a single repository. Mgit works on few repositories at the same time. That's the reason why you need split arguments for mgit (which touch e.g. scope of mgit) and command (that modified its behavior). 

I'll ask once again: should we remove `--` separator? If we want to have `mgit save --branch` and `mgit commit -- -m "Foo."`, how would like to remember which command needs `--` and which does not?

Reply:
> how would like to remember which command needs -- and which does not?

That's simple. `-m "Foo."` is git's param. Not mgit's. `--branch` is mgit's param, so it should be used without additional `--`.

Similarly, you can call `mgit diff -- --stat` – again, `--stat` is a git's param so I know that I don't want to pass it to `mgit diff` itself. 

The rule that you want to propose that some params like `--branch` or `--recursive` needs to be passed after additional `--` would require from us understanding how mgit works internally. 

> That's the reason why you need split arguments for mgit (which touch e.g. scope of mgit) and command (that modified its behavior).

There's no reason to split those arguments other than the implementation details. If mgit was implemented differently, you wouldn't think about this. In fact, so far `--recursive` was passed directly to mgit. You only changed it now.

Reply:
> The rule that you want to propose that some params like --branch or --recursive needs to be passed after additional -- would require from us understanding how mgit works internally.

No, I don't We just need to understand which params touch mgit itself and which executed commands. 

I never thought about `--` as you described: all before `--` touches mgit, all after `--` is applying to git.

But it does not matter. I'll do it as you want.